### PR TITLE
Handle contextual editors gracefully when restoring layout

### LINKED ISSUE #77013: Restarting editor with folded TileMap selected results in empty editor
### Godot version

4.0 fb10f45

### System information

Windows 10 x64

### Issue description

Behold, empty TileSet editor, with no TileMap node nor bottom button visible:
image
Regression from #72277 (most likely, I didn't test)

### Steps to reproduce

1. Add some Node to your scene
2. Add a TileMap as a child
3. Select TileMap
4. Fold its parent in the scene tree
5. Restart the editor
6. The TileMap selection is restored and so is the folded node, resulting in a weird paradox

### Minimal reproduction project

N/A

### LINKED ISSUE #77249: Crash when adding a new animation in AnimationFrame
### Godot version

master

### System information

PopOs 22.04, Nvidia

### Issue description

Under some conditions, adding a new animation to an AnimationFrame resource crashes the engine.
This happens only after restarting the engine, and trying to add a new animation on an AnimationFrame

https://github.com/godotengine/godot/assets/3624853/2586869b-bf0a-4840-a75a-f38f9c0bf649

Crash log : 

```
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.1.dev.custom_build (dc46163b121993229fedc7853000bd9bcb9ec2de)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7f1e1f242520] (??:0)
[2] HashMap<StringName, SpriteFrames::Anim, HashMapHasherDefault, HashMapComparatorDefault<StringName>, DefaultTypedAllocator<HashMapElement<StringName, SpriteFrames::Anim> > >::_lookup_pos(StringName const&, unsigned int&) const (/Projects/godot/./core/templates/hash_map.h:100)
[3] HashMap<StringName, SpriteFrames::Anim, HashMapHasherDefault, HashMapComparatorDefault<StringName>, DefaultTypedAllocator<HashMapElement<StringName, SpriteFrames::Anim> > >::has(StringName const&) const (/Projects/godot/./core/templates/hash_map.h:310)
[4] SpriteFrames::has_animation(StringName const&) const (/Projects/godot/scene/resources/sprite_frames.cpp:106)
[5] SpriteFramesEditor::_animation_add() (/Projects/godot/editor/plugins/sprite_frames_editor_plugin.cpp:1001)
[6] void call_with_variant_args_helper<SpriteFramesEditor>(SpriteFramesEditor*, void (SpriteFramesEditor::*)(), Variant const**, Callable::CallError&, IndexSequence<>) (/Projects/godot/./core/variant/binder_common.h:308)
[7] void call_with_variant_args<SpriteFramesEditor>(SpriteFramesEditor*, void (SpriteFramesEditor::*)(), Variant const**, int, Callable::CallError&) (/Projects/godot/./core/variant/binder_common.h:418)
[8] CallableCustomMethodPointer<SpriteFramesEditor>::call(Variant const**, int, Variant&, Callable::CallError&) const (/Projects/godot/./core/object/callable_method_pointer.h:105)
[9] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (/Projects/godot/core/variant/callable.cpp:51)
[10] Object::emit_signalp(StringName const&, Variant const**, int) (/Projects/godot/core/object/object.cpp:1068)
[11] Error Object::emit_signal<>(StringName const&) (/Projects/godot/./core/object/object.h:870)
[12] BaseButton::_pressed() (/Projects/godot/scene/gui/base_button.cpp:139)
[13] BaseButton::on_action_event(Ref<InputEvent>) (??:?)
[14] BaseButton::gui_input(Ref<InputEvent> const&) (/Projects/godot/scene/gui/base_button.cpp:69)
[15] Control::_call_gui_input(Ref<InputEvent> const&) (/Projects/godot/scene/gui/control.cpp:1752)
[16] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (/Projects/godot/scene/main/viewport.cpp:1511)
[17] Viewport::_gui_input_event(Ref<InputEvent>) (/Projects/godot/scene/main/viewport.cpp:1781)
[18] Viewport::push_input(Ref<InputEvent> const&, bool) (/Projects/godot/scene/main/viewport.cpp:2944)
[19] Window::_window_input(Ref<InputEvent> const&) (/Projects/godot/scene/main/window.cpp:1389)
[20] void call_with_variant_args_helper<Window, Ref<InputEvent> const&, 0ul>(Window*, void (Window::*)(Ref<InputEvent> const&), Variant const**, Callable::CallError&, IndexSequence<0ul>) (/Projects/godot/./core/variant/binder_common.h:303)
[21] void call_with_variant_args<Window, Ref<InputEvent> const&>(Window*, void (Window::*)(Ref<InputEvent> const&), Variant const**, int, Callable::CallError&) (/Projects/godot/./core/variant/binder_common.h:418)
[22] CallableCustomMethodPointer<Window, Ref<InputEvent> const&>::call(Variant const**, int, Variant&, Callable::CallError&) const (/Projects/godot/./core/object/callable_method_pointer.h:105)
[23] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (/Projects/godot/core/variant/callable.cpp:51)
[24] DisplayServerX11::_dispatch_input_event(Ref<InputEvent> const&) (/Projects/godot/platform/linuxbsd/x11/display_server_x11.cpp:3593)
[25] DisplayServerX11::_dispatch_input_events(Ref<InputEvent> const&) (/Projects/godot/platform/linuxbsd/x11/display_server_x11.cpp:3565)
[26] Input::_parse_input_event_impl(Ref<InputEvent> const&, bool) (/Projects/godot/core/input/input.cpp:705)
[27] Input::flush_buffered_events() (/Projects/godot/core/input/input.cpp:964)
[28] DisplayServerX11::process_events() (/Projects/godot/platform/linuxbsd/x11/display_server_x11.cpp:4661)
[29] OS_LinuxBSD::run() (/Projects/godot/platform/linuxbsd/os_linuxbsd.cpp:889)
[30] bin/godot.linuxbsd.editor.dev.x86_64.llvm(main+0x1fe) [0x55755e8189ae] (/Projects/godot/platform/linuxbsd/godot_linuxbsd.cpp:73)
[31] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7f1e1f229d90] (??:0)
[32] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7f1e1f229e40] (??:0)
[33] bin/godot.linuxbsd.editor.dev.x86_64.llvm(_start+0x25) [0x55755e8186e5] (??:?)
-- END OF BACKTRACE --

```

### Steps to reproduce

Open the MRP, and click `Add Animation`

### Minimal reproduction project

[Godot4Test.zip](https://github.com/godotengine/godot/files/11518538/Godot4Test.zip)

### LINKED ISSUE #78281: Theme editor starts opened and crashes upon interaction.
### Godot version

4.1 beta2

### System information

Windows 10 Home

### Issue description

I had a control with a custom theme which i removed. When loading the project again the Theme Editor is opened and some actions, like Show Default or Override All, cause the editor to crash.

### Steps to reproduce

1. create a scene with a Control node
2. add a custom theme to it
3. click the theme so the Theme Editor opens
4. remove the theme
5. save the scene
6. close the project and open it again (Theme Editor needs to be opened when closing the project)
7. click Show Default or Override All

### Minimal reproduction project

[Godot41_themeEditorCrash.zip](https://github.com/godotengine/godot/files/11760454/Godot41_themeEditorCrash.zip)

### PR DESCRIPTION
**Pull request by YuriSizov**
Reverts https://github.com/godotengine/godot/pull/78586 as unnecessary.

Still fixes https://github.com/godotengine/godot/issues/77249 and also still fixes https://github.com/godotengine/godot/issues/77013.
Fixes https://github.com/godotengine/godot/issues/78281.

While https://github.com/godotengine/godot/pull/78586 should prevent these invalid states from happening, it does so by simply ignoring contextual editors. And it doesn't handle existing stored layouts gracefully (so an already "corrupted" project can still result in these invalid states and lead to crashes). While looking into it, I figured this whole "permanent" trick wasn't necessary.

All we need to do, IMO, is check if the contextual editor is enabled (the button is visible on the bottom panel). If not, we just don't restore it. This handles both existing projects and future projects.

I also noticed that we don't remove stored information correctly about bottom panel and main screen editors. So I fixed it as well. https://github.com/godotengine/godot/pull/78586 introduced a similar problem, btw, where switching to a contextual editor and restarting the Godot editor would bring any previously stored "permanent" editor into focus. If we don't end up reverting it, this still would require fixing.

----

I noticed a couple of weird things which I don't think are related to either KoBeWi's or mine PR.

First, with the theme editor present I sometimes get `editor\editor_node.cpp:8199 - Condition "plugins_list.has(p_plugin)" is true.` on load (regardless if it's focused or not). It doesn't seem to lead to any problems.

Second, the MRP from https://github.com/godotengine/godot/issues/77249 had its `center_split_offset` permanently stored as -416. When you restored with a bottom panel editor opened, there were no visible problems. But if you closed all the editors and reloaded, the bottom panel would take too much space (about 416 pixels, or so :P) without any editor being open. And it wouldn't reset with consecutive reloads. Manually fixing the cached file to set it to 0 fixed the issue. But I wonder if there is an issue with the SplitContainer where it doesn't correctly reset its split value.

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]
Looks good to me. I'll let @KoBeWi review too before merging.

**Review by KoBeWi — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks for reviews!

**Pull request merged**
